FAIMS/FAIMS3
# feat: record progress bar

## JIRA Ticket

[BSS-383](https://jira.csiro.au/browse/BSS-383)

## Description

Add progress bar to top of record edit form to track completion of required fields.

## Proposed Changes

- add progress bar component
- update current form screens to include new component
- pass state updater to form to update progress
- create helper functions for measuring completion

## How to Test

1. Navigate to any survey
2. Click "+ NEW RECORD" button
3. Input values into required fields an check that progress bar reflects these changes
4. (optional) Publish and close record
5. (optional) Open published record and verify completion is retained

## Additional Information

The survey form seems reasonably unreliable and might be worth considering a rebuild at some point.
- the form re-renders around 11 times on load and another 5 times every few seconds (even without any changes)
- in testing I was able to get into a broken state where I would need to kill the tab and reload to resolve
- the code uses class components and is thousands of lines long across multiple files

## Checklist
